src/projects.h

```diff
@@ -597,14 +600,27 @@ C_NAMESPACE PJ *pj_##name (PJ *P) {                          \
         return 0;                                            \
     P->destructor = pj_default_destructor;                   \
     P->descr = des_##name;                                   \
-    P->left  = PJ_IO_UNITS_RADIANS;                          \
-    P->right = PJ_IO_UNITS_CLASSIC;                          \
+    P->need_ellps = NEED_ELLPS;                              \
+    P->left  = LEFT;                                         \
+    P->right = RIGHT;                                        \
     return P;                                                \
 }                                                            \
+                                                             \
 PJ *pj_projection_specific_setup_##name (PJ *P)
+
+/* In ISO19000 lingo, an operation is either a conversion or a transformation */
+#define CONVERSION(name, left, right, need_ellps)      OPERATION (name, left, right, need_ellps)
```

Comment:
Wouldn't it be simpler to assume that the units are meters on both sides for `CONVERSION`'s and `TRANSFORMATIONS`'s? In the rare case that something different is needed it can be changed in the operation setup.

Reply:
For most conversions it would be angular in, linear out, since most conversions are projections, and all projections are conversions, so in most cases the left and right arguments will be partially evaluated through the definition of the `PROJECTION` macro.

I actually think that direct calls to the `CONVERSION` macro will be rare, but it plays a role in making the code architecture reflect the conceptual model of ISO19000.

The transformations from the geodetic toolbox may be mostly linear in both ends, although e.g. Molodensky is angular in both ends, so I find it useful to have it comparatively clearly stated in the macro, (which amounts to a "kind of function header"), what kinds of I/O goes on.

Reply:
> I actually think that direct calls to the CONVERSION macro will be rare, but it plays a role in making the code architecture reflect the conceptual model of ISO19000.

`unitconvert` and `axisswap`, but yes, they are likely to be the only ones.

> The transformations from the geodetic toolbox may be mostly linear in both ends, although e.g. Molodensky is angular in both ends, so I find it useful to have it comparatively clearly stated in the macro, (which amounts to a "kind of function header"), what kinds of I/O goes on.

I don't feel too strongly about this either way, but I do prefer having a simple setup for most cases and paying a bit of extra attention in the few special cases like the Molodensky. 
Also, I think that the macro-part of the `PJ` initialisation is best reserved for things that are needed right away in `pj_init` such as `need_ellps` and the name of the operation. While important, I don't think the units fall in that category.